src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -290,26 +442,32 @@ def _create_binops_from_step_and_divisors(self, node, ref,
         binop2 = None
         if ref_index == 0:
             # We have Ref OP Literal
-            # Setup lambdas to do ref OP step when we then
+            # Setup vars to do ref OP step when we then
             # create the BinaryOperations to represent this
             # access
-            first_arg = lambda: ref.copy()
-            alt_first_arg = first_arg
-            second_arg = lambda: step
-            alt_second_arg = lambda: step2
+            first_arg = ref.copy()
+            alt_first_arg = first_arg.copy()
+            second_arg = step.copy()
+            if step2:
```

Comment:
It's a nicety but `alt_second_arg = step2.copy() if step2 else None` is a bit nicer and again at L461.

Reply:
Fixed